ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+                    ... code for workflow item 1 ...
+                case 2U:
+                    ... code for workflow item 2 ...
+                break;
+            }
+            return 2U;
+        }
+    };
+
+The main element is a switch statement inside the perform() function which allows the workflow engines to execute the code associated with a particular workflow item.
+
+There is also an associated factory function which is exported from the dll, and is used by the engines to create instances of the class::
+
+    extern "C" ECL_API IEclProcess* createProcess()
+    {
+        return new MyEclProcess;
+    }
+
+Graph
+=====
+
+Most of the execution of a query involves processing dataset operations.  These are represented in the workunit by xml
+graph structures.  (It currently uses xgmml format, although this may change.)  The graph xml includes details of which
+types of activities are required to be executed, how they are linked together, and any other dependencies.
+
+The graph in our example is particularly simple::
+
+  <Graph name="graph1" type="activities">
+   <xgmml>
+    <graph wfid="2">
+     <node id="1">
+      <att>
+       <graph>
+        <att name="rootGraph" value="1"/>
+        <edge id="2_0" source="2" target="3"/>
+        <node id="2" label="Index Read&#10;&apos;names&apos;">
+         ... attributes for activity 2 ...
+        </node>
+        <node id="3" label="Output&#10;Result #1">
+         ... attributes for activity 3 ...
+        </node>
+       </graph>
+      </att>
+     </node>
+    </graph>
+   </xgmml>
+  </Graph>
+
+This graph contains a single subgraph (node id=2) which contains two activities - an index read activity and an output result
+activity.  These activities are linked by a single edge (id "2_0").  The details of the contents are covered in the section on
+executing graphs below.
+
+Generated activity helpers
+==========================
+
+Each activity has a corresponding class instance in the generated code, and a factory function for creating instances of that class::
+
+    struct cAc2 : public CThorIndexReadArg {
+        ... Implementation of the helper for activity #2 ...
+    };
+    extern "C" ECL_API IHThorArg * fAc2() { return new cAc2; }
+    
+    struct cAc3 : public CThorWorkUnitWriteArg {
+        ... Implementation of the helper for activity #3 ...
+    };
+    extern "C" ECL_API IHThorArg * fAc3() { return new cAc3; }
+
+The helper class for an activity implements the interface that is required for that particular kind.  (The interfaces are defined in
+rtl/include/eclhelper.hpp - many more details on that below.)
+
+Other
+=====
+The are other items that are logically associated with a workunit.  They may be stored in the workunit dll or in
+various places in dali (or cassandra), but they are all accessed through the IWorkUnit interface in
+common/workunit/workunit.hpp which hides the implementation details.  For instance information generated at runtime cannot
```

Comment:
.hpp which hides the implementation details -> .hpp, hiding implementation details.
Sentence is too long and at least needs a comma.